x-pack/plugins/fleet/public/applications/fleet/sections/agents/agent_list_page/components/agent_activity_button.tsx

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiButtonEmpty, EuiText, EuiTourStep } from '@elastic/eui';
+import React, { useEffect, useState } from 'react';
+import { FormattedMessage } from '@kbn/i18n-react';
+
+import { useStartServices } from '../../../../hooks';
+
+export const AgentActivityButton: React.FC<{
+  onClickAgentActivity: () => void;
+  showAgentActivityTour: { isOpen: boolean };
+}> = ({ onClickAgentActivity, showAgentActivityTour }) => {
+  const { uiSettings } = useStartServices();
+
+  const [agentActivityTourState, setAgentActivityTourState] = useState(showAgentActivityTour);
+
+  const hideTour = () => uiSettings.get('hideAgentActivityTour', false);
+
+  const setHideTour = () => uiSettings.set('hideAgentActivityTour', true);
+
+  useEffect(() => {
```

Comment:
Syncing this state value to the value of a prop seems like it may be unnecessary. Can we just set this value on every render instead? 

https://beta.reactjs.org/learn/you-might-not-need-an-effect#updating-state-based-on-props-or-state

Reply:
the reason I've used a state here is that I close the tour from inside the component as well